### PR TITLE
KOGITO-1016: Merge Kogito examples and quickstarts repositories

### PR DESCRIPTION
**Pull request by evacchi**
**DO NOT SQUASH.** we want to maintain all the individual historical commits/contributors

When you review, pay most attention to the `pom.xml` and the main README.md. Code hasn't changed at all.

This is a preliminary PR to move this forward. We will do subsequent to align all the POMs and possibly change the layout of the repo. 

e.g.

- maybe it's better to have separate subdirs for springboot and quarkus ? 
- shall we have separate subdirs for "type" of example? e.g. hello world (old examples) and quickstart  and tutorials (travel agency, onboarding etc ?)
- shall we have separate profiles for CI and only run a subset of these for faster builds?

also, we should *definitely* point the default branch to the stable release (this can be easily done by changing the config of the repo)

### CODE REVIEW

**Comment by evacchi** [comment]
supersedes #139

File: kogito-business-rules-quarkus/.gitignore [hunk 1]
```diff
@@ -0,0 +1,20 @@
+bin/
```
Comment (cristianonicolai) [inline kogito-business-rules-quarkus/.gitignore]: @evacchi probably this file is not needed, keeping just the one at the root level?

File: kogito-business-rules-quarkus/pom.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    <version>8.0.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>kogito-business-rules-quarkus</artifactId>
+
+  <name>Kogito-business-rules-quarkus</name>
+  <description>Kogito business rules invocation - Quarkus</description>
+
+
+  <properties>
```
Comment (cristianonicolai) [inline kogito-business-rules-quarkus/pom.xml]: could be removed?

File: kogito-business-rules-springboot/.gitignore [hunk 3]
```diff
@@ -0,0 +1,20 @@
+bin/
```
Comment (cristianonicolai) [inline kogito-business-rules-springboot/.gitignore]: remove?

File: kogito-infinispan-persistence-quarkus/.gitignore [hunk 4]
```diff
@@ -0,0 +1,20 @@
+bin/
```
Comment (cristianonicolai) [inline kogito-infinispan-persistence-quarkus/.gitignore]: remove?

File: kogito-infinispan-persistence-quarkus/pom.xml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-openapi</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>infinispan-persistence-addon</artifactId>
+      <version>${kogito.version}</version>
```
Comment (cristianonicolai) [inline kogito-infinispan-persistence-quarkus/pom.xml]: version should be handled by the bom already.

File: kogito-infinispan-persistence-springboot/pom.xml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+  <properties>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.infinispan</groupId>
+        <artifactId>infinispan-bom</artifactId>
+        <version>10.0.0.Beta3</version>
```
Comment (cristianonicolai) [inline kogito-infinispan-persistence-springboot/pom.xml]: this is probably worth its own property at the root pom to keep all examples aligned.

File: kogito-infinispan-persistence-springboot/pom.xml [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.infinispan.protostream</groupId>
+        <artifactId>protostream</artifactId>
+        <version>4.3.0.Alpha8</version>
```
Comment (cristianonicolai) [inline kogito-infinispan-persistence-springboot/pom.xml]: same here

File: kogito-infinispan-persistence-springboot/pom.xml [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+      <groupId>org.infinispan</groupId>
+      <artifactId>infinispan-spring-boot-starter-remote</artifactId>
+      <version>${infinispan.version}</version>
+    </dependency>
+
+    <!-- kogito -->
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>kogito-springboot-starter</artifactId>
+      <version>${kogito.version}</version>
```
Comment (cristianonicolai) [inline kogito-infinispan-persistence-springboot/pom.xml]: already handled by bom?

File: kogito-kafka-quickstart-quarkus/.gitignore [hunk 9]
```diff
@@ -0,0 +1,20 @@
+bin/
```
Comment (cristianonicolai) [inline kogito-kafka-quickstart-quarkus/.gitignore]: remove?

**Review by cristianonicolai — COMMENTED** [review]
@evacchi some comments inline. Regarding your questions, here is my feedback:
e.g.
    _maybe it's better to have separate subdirs for springboot and quarkus ?_
I would leave as is for now.

_shall we have separate subdirs for "type" of example? e.g. hello world (old examples) and quickstart and tutorials (travel agency, onboarding etc ?)_
Same here, I think at least for now, we should leave it all flat. Perhaps review naming convention to make sure projects are probebly grouped.

_shall we have separate profiles for CI and only run a subset of these for faster builds?_
Not sure how would that work to be honest. If you change process it needs rules, if you change rules, it might break process. I think we could look into alternatives to speed up the build like maven cache, parallel builds, etc, but still running all examples.

_also, we should definitely point the default branch to the stable release (this can be easily done by changing the config of the repo)_
Thats the goal for sure, point master to the latest released version, and create a new branch ( dev/snapshot ? ) to work int the Jenkins pipeline. Just thinking that is probably best to address that in another pr? :)

File: kogito-business-rules-quarkus/pom.xml (hunk 2 above)
Comment (evacchi) [reply kogito-business-rules-quarkus/pom.xml]: I removed all of the empty `<properties>` sections now

File: kogito-infinispan-persistence-springboot/pom.xml (hunk 6 above)
Comment (evacchi) [reply kogito-infinispan-persistence-springboot/pom.xml]: I think there is no need for it, we already have it in our BOM. Removed

File: kogito-infinispan-persistence-springboot/pom.xml (hunk 8 above)
Comment (evacchi) [reply kogito-infinispan-persistence-springboot/pom.xml]: no need for it, already in our BOM. Removed

File: kogito-infinispan-persistence-springboot/pom.xml (hunk 7 above)
Comment (evacchi) [reply kogito-infinispan-persistence-springboot/pom.xml]: no need for it, already in our BOM. Removed

**Comment by evacchi** [comment]
>  Not sure how would that work to be honest. If you change process it needs rules, if you change rules, it might break process. I think we could look into alternatives to speed up the build like maven cache, parallel builds, etc, but still running all examples.

I just meant, maybe have all small examples run at once, while the "larger" examples (eg, travel agency, not yet here) being built on-demand.

>  Thats the goal for sure, point master to the latest released version, and create a new branch ( dev/snapshot ? ) to work int the Jenkins pipeline. Just thinking that is probably best to address that in another pr? :)

sure, just wanted to point it out already :-)

File: .circleci/config.yml [hunk 10]
```diff
@@ -0,0 +1,49 @@
+# Java Maven CircleCI 2.0 configuration file
```
Comment (cristianonicolai) [inline .circleci/config.yml]: @evacchi do we want to copy this over?

**Review by cristianonicolai — APPROVED** [review]
@evacchi just a minor comment, other that looks good.

File: .circleci/config.yml (hunk 10 above)
Comment (evacchi) [reply .circleci/config.yml]: no we don't. will remove :)

File: README.md [hunk 11]
```diff
@@ -6,4 +6,100 @@ This module contains a number of examples that you can take a look at and try ou
...
+
+# Kogito Quickstarts
+
+[![CircleCI](https://circleci.com/gh/mswiderski/kogito-quickstarts.svg?style=svg)](https://circleci.com/gh/mswiderski/kogito-quickstarts)
+
+A collection of quickstarts for Kogito that illustrates various uses case covered.
+
+Since Kogito aims at supporting both Quarkus and SpringBoot each quickstart should provide both type of projects.
+
+**To try it out locally on latest stable version download the released repository content [here](https://github.com/mswiderski/kogito-quickstarts/releases/latest)**
```
Comment (MarianMacik) [inline README.md]: I guess this is not needed anymore.

File: README.md [hunk 12]
```diff
@@ -6,4 +6,100 @@ This module contains a number of examples that you can take a look at and try ou
 - Drools on Quarkus: see [README.md](drools-quarkus-example/README.md)
 - Drools on Quarkus with Unit: see [README.md](drools-quarkus-unit-example/README.md)
 - jBPM on Quarkus: see [README.md](jbpm-quarkus-example/README.md)
-- jBPM on SpringBoot: see [README.md](jbpm-springboot-example/README.md)
\ No newline at end of file
+- jBPM on SpringBoot: see [README.md](jbpm-springboot-example/README.md)
+
+# Kogito Quickstarts
+
+[![CircleCI](https://circleci.com/gh/mswiderski/kogito-quickstarts.svg?style=svg)](https://circleci.com/gh/mswiderski/kogito-quickstarts)
```
Comment (MarianMacik) [inline README.md]: mswiderski reference is probably wrong here?

**Review by MarianMacik — APPROVED** [review]

**Comment by evacchi** [comment]
(squashed my commits)

**Review by radtriste — APPROVED** [review]
lgtm

**Comment by evacchi** [comment]
I have bumped to 1.3.0.CR2 but I would expect the BOM would have taken care of that. I'd prefer to merge this right away (as soon as it's green) and then we'll figure out how to pick it from BOM in a separate PR

**Comment by evacchi** [comment]
Jenkins please retest this

**Pull request merged**